From: wuchi <wuchi.zero@gmail.com>
Subject: initramfs: mark my_inptr as __initdata
Date: Sat, 27 Aug 2022 15:11:16 +0800

As my_inptr is only used in __init function unpack_to_rootfs(), mark it as
__initdata to allow it be freed after boot.

Link: https://lkml.kernel.org/r/20220827071116.83078-1-wuchi.zero@gmail.com
Signed-off-by: wuchi <wuchi.zero@gmail.com>
Cc: Alexander Viro <viro@zeniv.linux.org.uk>
Cc: David Disseldorp <ddiss@suse.de>
Cc: Martin Wilck <mwilck@suse.com>
Cc: Matthew Wilcox <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 init/initramfs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/init/initramfs.c~initramfs-mask-my_inptr-as-__initdata
+++ a/init/initramfs.c
@@ -482,7 +482,7 @@ static long __init flush_buffer(void *bu
 	return origLen;
 }
 
-static unsigned long my_inptr; /* index of next byte to be processed in inbuf */
+static unsigned long my_inptr __initdata; /* index of next byte to be processed in inbuf */
 
 #include <linux/decompress/generic.h>
 
_
